src/olympia/versions/tests.py

```diff
@@ -201,13 +201,13 @@ def test_version_delete(self):
 
         addon = Addon.objects.no_cache().get(pk=3615)
         assert not Version.objects.filter(addon=addon).exists()
-        assert not Version.unfiltered.filter(addon=addon).exists()
+        assert Version.unfiltered.filter(addon=addon).exists()
 
```

Comment:
I think we could add another explicit test here that tests the actual deletion (`hard=True`) on the model additionally to the test in the view. This way you would see both use-cases in model-related tests and the actual implementation is unrelated to the view too.